packages/editor/src/components/provider/index.js

```diff
@@ -301,15 +326,15 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 
 		// Sets the right rendering mode when loading the editor.
 		useEffect( () => {
-			setRenderingMode( settings.defaultRenderingMode ?? 'post-only' );
-		}, [ settings.defaultRenderingMode, setRenderingMode ] );
+			setRenderingMode( defaultMode );
+		}, [ defaultMode, setRenderingMode ] );
```

Comment:
This change is one that I'm not entirely sure about. I think the default rendering mode should only be enforced when the "edited entity" changes. In practice though, this seems to have the same effect. 

Anyway, not a big deal for now.